File: 백준/Silver/3273. 두 수의 합/두 수의 합.js
const fs = require('fs');
let [n, arr, x] = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
arr = arr.split(' ').map(Number);
x = Number(x);
arr.sort((a, b) => a - b);

let [left, right] = [0, n - 1];
let answer = 0;

while (left < right) {
    let sum = arr[left] + arr[right];
    if (sum < x) {
        left++;
    } else if (sum > x) {
        right--;
    } else {
        answer++;
        left++;
        right--;
    }
}

console.log(answer);

File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
import java.io.*;
import java.util.*;

public class Main {
    private static int answer;

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());
        int[] arr = new int[n];

        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(arr);

        int x = Integer.parseInt(br.readLine());

        int s = 0, e = n - 1;

        while (s < e){
            int sum = arr[s] + arr[e];
            if (sum > x){
                e --;
            }
            else if (sum < x){
                s ++;
            }
            else{
                answer ++;
                s++;
            }
        }
        System.out.println(answer);
    }
}